app/javascript/datasources/index.js

```diff
@@ -6,8 +6,35 @@ export const getPastMeetups = async () => {
     return json.data;
 };
 
-export const getJobs = async () => {
-    const result = await fetch(`${API_ROOT}/jobs`);
+export const getJobs = async (token) => {
+    const headers = new Headers();
+    headers.append('Authorization', `Bearer ${token}`);
+
+    const result = await fetch(`${API_ROOT}/jobs`, { headers });
+
+    if (result.status !== 200) {
+        throw new Error('Unauthorized');
+    }
```

Comment:
This is a new pattern I'm introducing -- if the query we perform is not successful, raise an error to reject the promise. I think this pattern can be improved further by returning descriptive error messages from the backend and then displaying those error messages in the frontend, thus handling a wider variety of errors, but for now I assume that if the query didn't work it's because the token was invalid.